src/ws_server/ws-server-handler.ts

```diff
@@ -0,0 +1,255 @@
+import { LoginReq, RegDataRes, UpdateWinDataRes } from '../models/player.js'
+import { MessageType, MyWebSocket, ResponseData, Server } from '../models/ws.js'
+import { generateResponse } from '../utils/generateResponse.js'
+import { stringifyWsMessage } from '../utils/stringifyWsMessage.js'
+import { PlayerToRoomDataReq, Room } from '../models/room.js'
+import { sendMessageToAllUsers } from '../utils/sendMessageToAllUsers.js'
+import {
+  AttackDataReq,
+  AttackDataRes,
+  Game,
+  PlayerTernDataRes,
+  RandomAttackDataReq,
+} from '../models/game.js'
+import { sendMessageToCurrentUser } from '../utils/sendMessageToCurrentUser.js'
+import { ShipReqData } from '../models/sheeps.js'
+import {
+  createGameObj,
+  createHitRegister,
+  createRoomObj,
+  getFirstPlayerData,
+  getStartGameData,
+  getUser,
+} from '../utils/crateObects.js'
+import { generateBattlefield } from '../utils/generateBattlefield.js'
+import { getAvailableCoordinates, hitRegistration } from '../utils/hitRegistration.js'
+
+export class WsServerHandler {
+  private users: RegDataRes[] = []
+  private rooms: Room[] = []
+  private games: Game[] = []
+  private winners: UpdateWinDataRes[] = []
+
+  constructor(private server: Server) {}
+
+  public registerUser = (message: LoginReq, ws: MyWebSocket) => {
+    const { name, password } = message.data
+    const user = getUser(name, password, ws.id, this.users)
+    if (!user) return
+    this.users.push(user)
+
+    const response = this.getResponse(message.type, user)
+    ws.send(response)
+    this.updateRoom()
+    this.sendWinners()
+  }
+
+  public createRoom = (ws: MyWebSocket) => {
+    const existedRoom = this.rooms.find(({ roomId }) => ws.id === roomId)
+    if (existedRoom) return
+
+    const user = this.users.find(({ index }) => ws.id === index)
+    if (!user) return
+
+    const room: Room = createRoomObj(ws.id, user.name, user.index)
+    this.rooms.push(room)
+
+    this.updateRoom()
+  }
+
+  updateRoom = () => {
+    const response = this.getResponse('update_room', this.rooms)
+    sendMessageToAllUsers(this.server, response)
```

Comment:
It's better to send notification to logged in users only (except the user who created that room).